From ccc893980fa85de2321ff01adde38ee0418da641 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:18 -0400
Subject: [tools] perf/trace: Beautify fcntl 'cmd' arg

Message-id: <1408462094-14194-90-git-send-email-jolsa@redhat.com>
Patchwork-id: 88073
O-Subject: [PATCH RHEL7.1 BZ1131394 089/325] perf trace: Beautify fcntl 'cmd' arg
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 80f587d5f9424ebdf93abbe79ab87e7fb7d4699d
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Sep 3 16:28:58 2013 -0300

description
===========
This is just for the low hanging fruit 'cmd' arg, a proper beautifier
will as well use arg->mask to ignore the third arg for some of the
cmds.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 449a9e7..cc57f41 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -244,6 +244,14 @@ static const char *whences[] = { "SET", "CUR", "END",
 };
 static DEFINE_STRARRAY(whences);
 
+static const char *fcntl_cmds[] = {
+ "DUPFD", "GETFD", "SETFD", "GETFL", "SETFL", "GETLK", "SETLK",
+ "SETLKW", "SETOWN", "GETOWN", "SETSIG", "GETSIG", "F_GETLK64",
+ "F_SETLK64", "F_SETLKW64", "F_SETOWN_EX", "F_GETOWN_EX",
+ "F_GETOWNER_UIDS",
+};
+static DEFINE_STRARRAY(fcntl_cmds);
+
 static size_t syscall_arg__scnprintf_open_flags(char *bf, size_t size,
             struct syscall_arg *arg)
 {
@@ -315,6 +323,9 @@ static struct syscall_fmt {
    .arg_scnprintf = { [0] = SCA_HEX, /* brk */ }, },
  { .name     = "mmap",     .hexret = true, },
  { .name     = "connect",    .errmsg = true, },
+ { .name     = "fcntl",     .errmsg = true,
+   .arg_scnprintf = { [1] = SCA_STRARRAY, /* cmd */ },
+   .arg_parm  = { [1] = &strarray__fcntl_cmds, /* cmd */ }, },
  { .name     = "fstat",     .errmsg = true, .alias = "newfstat", },
  { .name     = "fstatat",    .errmsg = true, .alias = "newfstatat", },
  { .name     = "futex",     .errmsg = true,
-- 
1.7.1